src/com/goide/inspections/GoStructInitializationInspection.java

```diff
@@ -36,8 +36,11 @@ public void visitCompositeLit(@NotNull GoCompositeLit o) {
         if (PsiTreeUtil.getParentOfType(o, GoReturnStatement.class, GoShortVarDeclaration.class, GoAssignmentStatement.class) == null) return;
 
         GoType type = o.getType();
-        if (!(type instanceof GoStructType) &&
-            !(GoPsiImplUtil.findBaseTypeFromRef(o.getTypeReferenceExpression()) instanceof GoSpecType)) {
+        GoType typeFromTypeRef = GoPsiImplUtil.findTypeFromTypeRef(o.getTypeReferenceExpression());
+        if (!(type instanceof GoStructType ||
```

Comment:
I'll restore the changes. As for the quickfix, we have: https://github.com/go-lang-plugin-org/go-lang-idea-plugin/issues/2137